src/ol_infrastructure/applications/jupyterhub/__main__.py

```diff
@@ -76,6 +91,68 @@
     lambda ns: check_cluster_namespace(namespace, ns)
 )
 
+
+rds_defaults = defaults(stack_info)["rds"]
+rds_defaults["instance_size"] = (
+    jupyterhub_config.get("db_instance_size") or rds_defaults["instance_size"]
+)
+rds_defaults["use_blue_green"] = False
+rds_password = jupyterhub_config.require("rds_password")
+
+jupyterhub_db_security_group = ec2.SecurityGroup(
+    f"jupyterhub-db-security-group-{env_name}",
+    name=f"jupyterhub-db-{target_vpc_name}-{env_name}",
+    description="Access from jupyterhub to its own postgres database.",
+    ingress=[
+        ec2.SecurityGroupIngressArgs(
+            security_groups=[
+                vault_stack.require_output("vault_server")["security_group"],
+            ],
+            cidr_blocks=[target_vpc["cidr"]],
+            protocol="tcp",
+            from_port=DEFAULT_POSTGRES_PORT,
+            to_port=DEFAULT_POSTGRES_PORT,
+            description="Access to Postgres from jupyterhub nodes.",
+        ),
+        ec2.SecurityGroupIngressArgs(
+            cidr_blocks=k8s_pod_subnet_cidrs,
+            description="Allow k8s cluster ipblocks to talk to DB",
+            from_port=DEFAULT_POSTGRES_PORT,
+            protocol="tcp",
+            security_groups=[],
+            to_port=DEFAULT_POSTGRES_PORT,
+        ),
+    ],
+    tags=aws_config.tags,
+    vpc_id=target_vpc_id,
+)
+
+jupyterhub_db_config = OLPostgresDBConfig(
+    instance_name=f"jupyterhub-db-{stack_info.env_suffix}",
+    password=rds_password,
+    storage=jupyterhub_config.get("db_capacity")
+    or str(AWS_RDS_DEFAULT_DATABASE_CAPACITY),
+    subnet_group_name=apps_vpc["rds_subnet"],
+    security_groups=[jupyterhub_db_security_group],
+    parameter_overrides=[{"name": "rds.force_ssl", "value": 0}],
+    engine_major_version="16",
```

Comment:
```suggestion
```
Where possible I like to let this delegate to the component resource to determine that version